### PR TITLE
KNOX-2625 - initial commit

### PR DESCRIPTION
**Pull request by lmccay**
Change-Id: I042bed7d6cedccf33a8880b604fd2211b541db66

(It is very **important** that you created an Apache Knox JIRA for this change and that the PR title/commit message includes the Apache Knox JIRA ID!)

## What changes were proposed in this pull request?

Logout page for session timeout and/or logout flows.
This is initial patch and will need a couple more iterations but good for testing and
dev of rewrite rules for participating applications for rewriting Location header that
redirects to their own logout landing page.

## How was this patch tested?

Manually tested with the logout button on the Knox homepage

Please review [Knox Contributing Process](https://cwiki.apache.org/confluence/display/KNOX/Contribution+Process#ContributionProcess-GithubWorkflow) before opening a pull request.

### CODE REVIEW

File: knox-homepage-ui/home/app/sessionInformation/session.information.component.ts [hunk 1]
```diff
@@ -35,10 +35,11 @@ export class SessionInformationComponent implements OnInit {
 
     getUser() {
         if (this.sessionInformation) {
-            console.debug('SessionInformationComponent --> getUser() --> ' + this.sessionInformation.user);
-            return this.sessionInformation.user;
+          console.debug('SessionInformationComponent --> getUser() --> ' + this.sessionInformation.user);
```
Comment (moresandeep) [inline knox-homepage-ui/home/app/sessionInformation/session.information.component.ts:38]: Same as above.

**Review by moresandeep — APPROVED** [review]
I have minor comments but other than that looks good.

File: gateway-applications/src/main/resources/applications/knoxauth/app/logout.jsp [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        Collection<Service> services = topology.getServices();
+        for (Object service : services) {
+          Service svc = (Service)service;
+          if (svc.getRole().equals("KNOXSSO")) {
+            Map<String, String> params = svc.getParams();
+            whitelist = params.get("knoxsso.redirect.whitelist.regex");
+            // LJM TODO: get cookie name and possibly domain prefix info for use in logout
+            cookieName = params.get("knoxsso.cookie.name");
+            if (cookieName == null) {
+                cookieName = "hadoop-jwt";
```
[the remark this answers is not in the document]
Comment (lmccay) [reply gateway-applications/src/main/resources/applications/knoxauth/app/logout.jsp:67]: I don't think so. We are checking the configured cookie name here and setting to hadoop-jwt if not configured - right?

[the remark this answers is not in the document]
Comment (moresandeep) [reply gateway-applications/src/main/resources/applications/knoxauth/app/logout.jsp:67]: Darn it, you are right!

**Pull request merged**